Feature/#5

JJJoonngg opened the pull request.

JJJoonngg committed:
#5 kakao 로그인 임시 구현

- splash screen 로직과 합치기 위해서 commit
- kakao app key 값은 별도로 관리
- login button 임시 추가

JJJoonngg committed:
Merge branch 'develop' into feature/#5

# Conflicts:
#	NanDoChak/app/src/main/AndroidManifest.xml
#	NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/util/NanDoChakApplication.kt

JJJoonngg committed:
Merge branch 'develop' into feature/#5

JJJoonngg committed:
Merge branch 'develop' into feature/#5

JJJoonngg committed:
#5 login 화면 페이지 구성

- 앱 메인 color 변경
- 로그인 화면 구성

JJJoonngg committed:
#5 kakao login 구현

- 카카오 톡이 존재하면 카카오톡으로, 존재하지 않는다면 이를 카카오 계정을 통해서 진행하도록 수행
- 유저의 토큰값을 이용하여 회원 여부를 체크 예정

JJJoonngg committed:
#5 nick name input 화면 구현

- enter 를 눌렀을 경우 다음 화면으로 넘어가도록 구현
- intent 를 통해서 유저 닉네임을 전달하도록 구현

JJJoonngg committed:
#5 유저 닉네임 입력 이후 평균 준비 시간으로 넘어오도록 로직 구현

JJJoonngg committed:
#5 온보딩 페이지 구현 완료

JJJoonngg committed:
#5 카카오 로그인 구현
#25 Firestore accessor 구현

- user token 이 아닌 고유한 값인 id 를 이용하여 kakao login 구현 및  firestore 액세서 구현
- 로그인 실패의 경우에는 토스트가 뜨도록 설정
- 서버와 통신하는 시간이 있기 때문에 로딩 화면이 있어야 한다는 판단

gyurim commented on NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/firebase/FirestoreAccessor.kt:
```diff
@@ -0,0 +1,90 @@
+
+    const val FIRESTORE_READ_TIMEOUT = 60L * 1000L
+
+    enum class UserExistCheck {
```
오홍 저는 enum class는 항상 파일로 뺐는데 이렇게 하는 방법도 있군용..b  
더 편하게 볼 수 있을 거 같아요!

gyurim commented on NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/view/signup/SignUpActivity.kt:
```diff
@@ -0,0 +1,89 @@
+
+    private fun startAverageActivity(nickName: String) {
+        Intent(this, AverageReadyTimeActivity::class.java).apply {
+            putExtra(AppConfig.INTENT_PARAM_NICK_NAME, nickName)
```
궁금한 것 있습니다! 
nickname을 intent로 넘기는 것과 DB에서 nickname을 읽어오는 것 중 intent가 성능이 더 나아서 해당 방법 사용한 것인지 궁금해요! 
nickname을 계속 intent로 넘기던데 더 좋은 방법이 없을까요 ㅜㅜ ? 매번 플젝을 하면서 항상 궁금했습니당.. 중복 intent 줄일 수 있는 법 ..

JJJoonngg replied:
회원 가입 로직에서는 굳이 db 에 저장하지 않고 intent 만 받아서 쓰려고 해서 해당 로직을 구현했습니다
singleton 에서 값을 가져올까 하다가 일단은 nickname 으로 두었고, 뒤에 로직은 singleton 으로 값을 가져오게 합니다!

gyurim replied:
아하 설명 감사합니다!!

gyurim commented on NanDoChak/app/src/main/res/layout/activity_early_arrived_time.xml:
```diff
@@ -17,80 +18,80 @@
         <TextView
             android:id="@+id/earlyArrivedTimeTitle"
-            android:text="@string/selectEarlyArrivedTime"
+            style="@style/TitleTextView"
```
허헣.. 쪼개기.. 수고하셨습니다 ;_; b

gyurim commented on NanDoChak/app/src/main/java/kr/co/west_gang/nan_do_chak/view/signup/SignUpActivity.kt:
```diff
@@ -0,0 +1,89 @@
+                Unit
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                val input = s?.toString()?.replace(" ", "") ?: ""
```
edittext도 data binding으로 진행하면 어떨까요?

https://apt-info.github.io/%EA%B0%9C%EB%B0%9C/android-mvvm-edittext/ 
제가 참고한 링크는 이거예요!

JJJoonngg replied:
링크 주신 inverse databinding 으로 해서 구현해 볼게요 ㅎㅎ

JJJoonngg committed:
#5 리뷰반영

- two-way binding(edit text 기본 제공) 이용
- activity result 이용해서 기존에 열려있던 activity 종료하도록 변경
